@histoire/plugin-svelte: Peer Dependency "svelte": "^3.0.0" prevents installation of svelte 4

### Describe the bug

When migrating from svelte 3 to svelte 4 in a sveltekit project which uses histoire, I get the following error on npm install.

```shell
$ npm i
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: @histoire/plugin-svelte@0.16.1
npm ERR! Found: svelte@4.0.0
npm ERR! node_modules/svelte
npm ERR!   dev svelte@"^4.0.0" from the root project
npm ERR!   peer svelte@"^3.54.0 || ^4.0.0-next.0" from @sveltejs/kit@1.20.4
npm ERR!   node_modules/@sveltejs/kit
npm ERR!     dev @sveltejs/kit@"^1.20.4" from the root project
npm ERR!     peer @sveltejs/kit@"^1.0.0" from @sveltejs/adapter-auto@2.1.0
npm ERR!     node_modules/@sveltejs/adapter-auto
npm ERR!       dev @sveltejs/adapter-auto@"^2.0.0" from the root project
npm ERR!     1 more (@sveltejs/adapter-static)
npm ERR!   6 more (@sveltejs/vite-plugin-svelte, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer svelte@"^3.0.0" from @histoire/plugin-svelte@0.16.1
npm ERR! node_modules/@histoire/plugin-svelte
npm ERR!   dev @histoire/plugin-svelte@"^0.16.1" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: svelte@3.59.2
npm ERR! node_modules/svelte
npm ERR!   peer svelte@"^3.0.0" from @histoire/plugin-svelte@0.16.1
npm ERR!   node_modules/@histoire/plugin-svelte
npm ERR!     dev @histoire/plugin-svelte@"^0.16.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

```

So my conclusion is: @histoire/plugin-svelte: Peer Dependency "svelte": "^3.0.0" prevents installation of svelte 4

### Reproduction

Install sveltekit in version ^1.13.0 with svelte@3 with @histoire/plugin-svelte and then run `npx svelte-migrate svelte-4` to migrate to svelte@4 and sveltekit to ^1.20.4.

This will lead to conflicting peer dependencies because of peer dependency definition of @histoire/plugin-svelte@0.16.1 .

### System Info

```shell
System:
    OS: Linux 5.15 Ubuntu 22.04.2 LTS 22.04.2 LTS (Jammy Jellyfish)
    CPU: (16) x64 AMD Ryzen 7 PRO 5850U with Radeon Graphics
    Memory: 27.08 GB / 30.18 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 18.16.1 - ~/.nvm/versions/node/v18.16.1/bin/node
    npm: 9.7.2 - ~/.nvm/versions/node/v18.16.1/bin/npm
  Browsers:
    Brave Browser: 114.1.52.126
  npmPackages:
    @histoire/plugin-svelte: ^0.16.1 => 0.16.1 
    histoire: ^0.16.1 => 0.16.2 
    vite: ^4.2.1 => 4.3.9
```


### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/histoire-dev/histoire/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/histoire-dev/histoire/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://histoire.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/histoire-dev/histoire/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/histoire-dev/histoire/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/examples/svelte3/package.json
+++ b/examples/svelte3/package.json
@@ -13,15 +13,15 @@
   },
   "type": "module",
   "dependencies": {
-    "svelte": "^3.54.0"
+    "svelte": "^4.1.2"
   },
   "devDependencies": {
     "@histoire/plugin-svelte": "workspace:*",
-    "@sveltejs/vite-plugin-svelte": "^2.0.0",
+    "@sveltejs/vite-plugin-svelte": "^2.4.3",
     "cypress": "^10.8.0",
     "histoire": "workspace:*",
     "start-server-and-test": "^1.14.0",
-    "svelte-preprocess": "^4.10.7",
+    "svelte-preprocess": "^5.0.4",
     "vite": "^4.2.0"
   }
 }

--- a/packages/histoire-plugin-svelte/package.json
+++ b/packages/histoire-plugin-svelte/package.json
@@ -55,12 +55,12 @@
     "fs-extra": "^10.0.1",
     "globby": "^13.1.1",
     "histoire": "workspace:*",
-    "svelte": "^3.49.0",
+    "svelte": "^3.49.0 || ^4.0.0",
     "svelte-preprocess": "^4.10.7",
     "vite": "^4.2.0"
   },
   "peerDependencies": {
     "histoire": "workspace:^",
-    "svelte": "^3.0.0"
+    "svelte": "^3.0.0 || ^4.0.0"
   }
 }

